src/lib.rs

```diff
@@ -403,6 +403,13 @@ where
         self.get(key, guard).map(then)
     }
 
+    /// Returns the key-value pair corresponding to the supplied key.
+    ///
+    /// Returns `None` if this map contains no mapping for `key`.
+    pub fn get_key_value<'g>(&'g self, key: &'g K, guard: &'g Guard) -> Option<(&'g K, &'g V)> {
```

Comment:
I don't think you need the `'g` on the `self` reference here. As long as the guard lives, the pointers will remain valid, even if `self` completely goes away.

Reply:
In fact, I don't think you need it on `key` either, since we return the key we read from the map.

Reply:
Actually, damn, I take that back. I think there's a bug in `insert`. The `'g` _does_ have to be associated with the lifetime of `self` for [this](https://github.com/jonhoo/flurry/blob/6d56c58026f6edd0d9828ce48b3cbb6aecd38973/src/lib.rs#L1227) to be safe. It shouldn't be needed on `key` though!

Reply:
Ah, I see, below you _are_ returning the `key` that is passed in. That is probably not what we want to do. Instead, we'll want an internal version of `get` that produces a `&Node`, and then have `get` call that and extract the key, and this have call that and extract both the key and the value.